packages/ember-metal/lib/properties.js

```diff
@@ -139,58 +160,41 @@ export function defineProperty(obj, keyName, desc, data, meta) {
   watching = watchEntry !== undefined && watchEntry > 0;
 
   if (existingDesc) {
-    existingDesc.teardown(obj, keyName);
+    existingDesc.teardown(obj, keyName, watching, meta);
   }
 
-  if (desc instanceof Descriptor) {
-    value = desc;
+  if (desc === null || desc === undefined) {
```

Comment:
This branch was just moved up, mostly unchanged. The condition was previously `if (desc == null)` – note the _double_ equal.